fairscale/experimental/wgit/pygit.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+from pathlib import Path
+from typing import List
+
+import pygit2
+
+
+class PyGit:
+    def __init__(
+        self,
+        parent_path: Path,
+        gitignore: List = list(),
+        name: str = "user",
+        email: str = "user@email.com",
+    ) -> None:
+        """
+        PyGit class to wrap the wgit/.git repo and interact with the git repo.
+
+        Args:
+        parent_path: Has to be the full path of the parent!
+        """
+        # Find if a git repo exists within .wgit repo:
+        # If exists: then discover it and set the self.gitrepo path to its path
+        self._parent_path = parent_path
+        self.name = name
+        self.email = email
+
+        git_repo_found = pygit2.discover_repository(self._parent_path)
+
+        if git_repo_found:
+            # grab the parent dir of this git repo
+            git_repo = Path(pygit2.discover_repository(self._parent_path))
+            pygit_parent_p = git_repo.parent.absolute()
+
+            # Check If the parent dir is a .wgit dir. If the .wgit is a git repo
+            # just wrap the existing git repo with pygit2.Repository class
+            if pygit_parent_p == self._parent_path:
+                self.repo = pygit2.Repository(str(self._parent_path))
+                self.path = self._parent_path.joinpath(".git")
+            else:
+                # if the parent is not a .wgit repo,
+                # then the found-repo is a different git repo. Init new .wgit/.git
+                self._init_wgit_git(gitignore)
+        else:
+            # no git repo found, make .wgit a git repo
+            self._init_wgit_git(gitignore)
+
+    def _init_wgit_git(self, gitignore: List) -> None:
```

Comment:
document the gitignore arg. you can use "see above" since it is the same as that in the `__init__`.